kafka-connect/kafkaconnect-w-auth.yml

```diff
@@ -0,0 +1,79 @@
+kind: Template
+metadata:
+  name: kafka-connect-template
+objects:
+- apiVersion: kafka.strimzi.io/v1beta2
+  kind: KafkaConnect
+  metadata:
+    annotations:
+      strimzi.io/use-connector-resources: "true"
+    name: ${SERVICE_NAME}-kafka-connect
+  spec:
+    bootstrapServers: ${BOOTSTRAP_SERVERS}
+    authentication:
+      type: scram-sha-512
+      username: ${KAFKA_USERNAME}
+      passwordSecret:
+        secretName: ${KAFKA_USER_SECRET_NAME}
+        password: ${KAFKA_USER_SECRET_KEY}
+    config:
+      config.storage.replication.factor: ${CONFIG_STORAGE_REPLICATION_FACTOR}
+      config.storage.topic: connect-cluster-configs
+      connector.client.config.override.policy: All
+      group.id: ${SERVICE_NAME}_connect-cluster
+      offset.storage.replication.factor: ${OFFSET_STORAGE_REPLICATION_FACTOR}
+      offset.storage.topic: connect-cluster-offsets
+      status.storage.replication.factor: ${STATUS_STORAGE_REPLICATION_FACTOR}
+      status.storage.topic: connect-cluster-status
+      config.providers: secrets
+      config.providers.secrets.class: io.strimzi.kafka.KubernetesSecretConfigProvider
+    image: ${KAFKA_CONNECT_IMAGE}
+    replicas: ${{REPLICAS}}
+    resources:
+      limits:
+        cpu: 500m
+        memory: 1Gi
+      requests:
+        cpu: 250m
+        memory: 512Mi
+    template:
+      pod:
+        imagePullSecrets:
+        - name: quay-cloudservices-pull
+    version: ${VERSION}
+parameters:
+  - name: SERVICE_NAME
+    description: Name of the service the connect cluster serves (used as prefix in naming)
+    required: true
+  - name: BOOTSTRAP_SERVERS
+    description: List of bootstrap servers (comma-separated list in 'hostname:port' notation)
+    required: true
+  - name: KAFKA_USERNAME
+    description: Kafka Username used for the authentication
+    required: true
+  - name: KAFKA_USER_SECRET_NAME
+    description: The name of the Secret containing the password for the Kafka User
+    required: true
+  - name: KAFKA_USER_SECRET_KEY
+    description: The name of the key in the Secret under which the password is stored
+    required: true
+    value: password
+  - name: CONFIG_STORAGE_REPLICATION_FACTOR
+    description: Replication factor for the topic where connector configurations are stored
+    value: "1"
+  - name: OFFSET_STORAGE_REPLICATION_FACTOR
+    description: Replication factor for the topic where source connector offsets are store
+    value: "1"
+  - name: STATUS_STORAGE_REPLICATION_FACTOR
+    description: Replication factor for the topic where connector and task status are stored
+    value: "1"
+  - name: KAFKA_CONNECT_IMAGE
+    description: Container image name for the connect cluster pods
+    required: true
+  - name: REPLICAS
+    description: Number of replicas in the connect cluster
+    value: "1"
+  - name: VERSION
+    description: Kafka Connect version to use (should match the Kafka version of cluster and connect base image)
+    value: "3.9.0"
```

Comment:
I wonder if we should recommend setting version in the `oc process` command since it sounds like the versions are coupled pretty tightly.

Reply:
Yeah probably not a bad idea, i put 3.9.0 as thats a sane default that will work with current infra but anything that ensures users are aware of the setting and makes them think about/ask what should it be is probably good for long term educational purposes